articles/20240820_optimizing-gpu-utilization-in-containerized-environments.md

```diff
@@ -0,0 +1,271 @@
+Workspace Data Dir
+The directory on the remote host where the workspace data will be stored
+> /tmp/daytona-data
+```
+Click `Enter` to confirm, and you should see a confirmation message indicating that the `Target was set successfully`.
+
+## Verifying GPU Setup
+
+## Install the NVIDIA Container Toolkit
+
+  Install the NVIDIA Container Toolkit to enable GPU support in Docker:
+
+```bash
+sudo apt-get update && sudo apt-get install -y nvidia-container-toolkit
+sudo systemctl restart docker
+```
+This toolkit allows Docker to interface directly with your GPU hardware, a crucial step for GPU-intensive tasks.
+
+### Verify Your GPU Setup
+
+  Confirm that Docker can access your GPU:
+
+```bash
+docker run --gpus all nvidia/cuda:11.0-base nvidia-smi
+```
+This command verifies that your GPU is accessible from within Docker containers.
+
+## Step 2: Setting Up a Workspace in Daytona
+
+For this guide, we will use [nanoGPT](https://github.com/karpathy/nanoGPT) as our example project. nanoGPT is a lightweight implementation of GPT, making it an best example for demonstrating how to configure and run GPU-intensive tasks in a containerized environment.
+
+### Create a Workspace with the Git Repository:
+
+Daytona workspaces serve as isolated environments containing your projects, containing the codebase, dependencies, packages, and configurations. They ensure consistency across different development, testing, and deployment.
+
+#### Creating a Workspace from a Git Repository:
+
+To directly set up a [workspace from a Git repository]( https://www.daytona.io/docs/usage/workspaces/#create-a-workspace) like nanoGPT, Daytona makes the process straightforward:
+
+1. **Run the Command**:
+
+```bash
+daytona create https://github.com/karpathy/nanoGPT.git
+```
+
+This command will clone the nanoGPT repository and set up a new workspace with the project ready for development.
+
+2. **Workspace Setup**:
+
+Daytona manages the initialization and configuration of the workspace, displaying progress information in the terminal. Once completed, the workspace will be ready for use. Example output during workspace creation:
+
+```plaintext
+WORKSPACE | ✓ Request submitted
+WORKSPACE | ✓ Creating workspace nanoGPT
+daytona   | Creating project nanoGPT
+daytona   | Pulling image...
+daytona   | Pulling from daytonaio/workspace-project
+...
+daytona   | Project nanoGPT created
+daytona   | Starting project nanoGPT
+daytona   | Project nanoGPT started
+```
+
+#### List Workspaces:
+
+To keep track of all your workspaces:
+
+```bash
+daytona list
+```
+
+This command lists all created workspaces, including their repository, target, and status.
+
+**Tip**: You can open the created nanoGPT workspace in your preferred [IDE](https://www.daytona.io/docs/reference/cli/#daytona-code) `daytona code` command
+
+## Build and Run the Docker Image
+
+### Create a Dockerfile for nanoGPT
+
+Next, create a Dockerfile tailored to run nanoGPT with GPU support. Here’s an example Dockerfile:
```

Comment:
I suggest you create a project, push to github, do `daytona create your-repo-url`, and share screenshots or steps to replicate yout project.

you can easily define the dependencies or packages devconatiner.json file.